### PR TITLE
Automatic user authorization

### PR DESCRIPTION
**Pull request by Faz-R**
* Add a record to the isAutoLogin field in the redux store with saving its value in local storage using redux-persist
* Add a record to this field when the “Remember me” checkbox is activated in case of successful authorization
* Add a request to MainHeader to validate the authorization token
* If the response to the request to validate the authorization token is 200 and the isAutoLogin field=true, then we make a transition to the Income page

### CODE REVIEW

File: src/components/mainLayout/mainHeader/mainHeader.tsx [hunk 1]
```diff
@@ -30,6 +41,26 @@ const MainHeader = () => {
...
+	}, []);
+
+	useEffect(() => {
+		try {
+			if (baseUrl) {
+				validateToken(baseUrl).then(() => {
+					if (loginStatus) {
+						return router.push(UserProfilePath.ProfitMoney);
+					}
+				});
```
Comment (AlexMakovetskiy) [inline src/components/mainLayout/mainHeader/mainHeader.tsx:55]: validateToken(baseUrl).then((response: AxiosResponse<any, any>) => {
					if (loginStatus && response.status === HttpStatusCode.Ok) {
						return router.push(UserProfilePath.ProfitMoney);
					}
				});

Comment (Faz-R) [reply src/components/mainLayout/mainHeader/mainHeader.tsx:55]: I have error with HttpStatusCode import. So I switched off eslint rules // eslint-disable-next-line import/named

Comment (AlexMakovetskiy) [reply src/components/mainLayout/mainHeader/mainHeader.tsx]: Just press button "Disable eslint rule for current line"

File: src/services/redux/index.ts [hunk 2]
```diff
@@ -11,13 +11,15 @@ const persistConfig = {
...
 
 const store = configureStore({
-	reducer: {
-		data: dataReducer,
-		status: persistedStatusReducer,
-	},
+	reducer: persistedReducer,
 	middleware: (getDefaultMiddleware) =>
 		getDefaultMiddleware({
 			serializableCheck: { ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER] },
```
Comment (AlexMakovetskiy) [inline src/services/redux/index.ts:25]: const persistConfig = {
	key: "root",
	storage,
	whitelist: ["status"],
};

const persistedStatusReducer = persistReducer(persistConfig, statusReducer);

const store = configureStore({
	reducer: {
		data: dataReducer,
		status: persistedStatusReducer,
	},
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: { ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER] },
		}),
});

File: src/services/redux/index.ts [hunk 3]
```diff
@@ -11,13 +11,15 @@ const persistConfig = {
...
+})
+
+const persistedReducer = persistReducer(persistConfig, rootReducer)
 
 const store = configureStore({
-	reducer: {
-		data: dataReducer,
-		status: persistedStatusReducer,
-	},
+	reducer: persistedReducer,
```
Comment (AlexMakovetskiy) [inline src/services/redux/index.ts:22]:
```

const persistConfig = {
	key: "root",
	storage,
	whitelist: ["status"],
};

const persistedStatusReducer = persistReducer(persistConfig, statusReducer);

const store = configureStore({
	reducer: {
		data: dataReducer,
		status: persistedStatusReducer,
	},
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: { ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER] },
		}),
});
```

**Review by AlexMakovetskiy — APPROVED** [review]

**Pull request merged**